### DISCUSSION TITLE
Control permissions does not allow adding workflow files

### DISCUSSION THREAD

**Discussion by stefanbuck** [community/community#26583] [Actions]
<p>With excitement I read the latest addition to <a href="https://github.blog/changelog/2021-04-20-github-actions-control-permissions-for-github_token/" rel="noopener nofollow ugc">control the permissions granted to the <code>GITHUB_TOKEN</code> secret.</a></p>
<p>I do have a workflow that adds another workflow file to a repository. Up until now, I use a custom Personal Access Token with <code>workflows</code> permission in order to achieve that. I was hoping that this addition would allow me to get rid of this PAT by using the following permissions.</p>
<pre><code class="lang-auto">permissions:
  actions: write
  contents: write
</code></pre>
<p>but it turns out it doesn’t work.</p>
<pre><code class="lang-auto">[remote rejected] temp-branch -&gt; main (refusing to allow a GitHub App to create or update workflow `.github/workflows/runner-4.yml` without `workflows` permission)

error: failed to push some refs to 'https://github.com/....'
</code></pre>
<p>I wonder if this is expected behavior, given that I set <code>actions</code> to <code>write</code> or do I misinterpret this permission?</p>

**Comment by otter-computer** [comment, marked as answer]
<p>Hi <a class="mention" href="/stefanbuck">@stefanbuck</a>! I believe this is expected behaviour for this permission. Changing these permissions allows you to scope how much you can interact with the <a href="https://docs.github.com/en/rest/reference/permissions-required-for-github-apps">API of the underlying GitHub App</a>, but this app doesn’t have the scope required to update workflow files without the <code>workflow</code> permission, which isn’t currently configurable here. The <a href="https://docs.github.com/en/rest/reference/permissions-required-for-github-apps#permission-on-actions">actions permission</a> currently allows you to interact with various aspects of workflows on your repository, but doesn’t allow you to update workflow files.</p>
<p>If you need your workflow to be able to create or modify workflow files then you’ll need to continue to use a PAT that has the <code>workflow</code> permision.</p>
